### PR TITLE
bpo-30319: socket.close() now ignores ECONNRESET

### PR DESCRIPTION
**Pull request by vstinner**
http://bugs.python.org/issue30319

### CODE REVIEW

**Comment by pitrou** [comment]
Please dissociate the close() and shutdown() changes.

**Comment by vstinner** [comment]
> Please dissociate the close() and shutdown() changes.

Done.

File: Doc/library/socket.rst [hunk 1]
```diff
@@ -928,6 +928,9 @@ to sockets.
       :exc:`OSError` is now raised if an error occurs when the underlying
       :c:func:`close` call is made.
 
+   .. versionchanged:: 3.7
```
Comment (pitrou) [inline Doc/library/socket.rst]: I think it's misleading to add a "versionchanged" here. It's just a regression that we are fixing.

**Comment by pitrou** [comment]
I think that this also needs a 3.6 backport label.

File: Doc/library/socket.rst (hunk 1 above)
Comment (vstinner) [reply Doc/library/socket.rst]: removed

**Comment by vstinner** [comment]
> I think that this also needs a 3.6 backport label.

I agree, I added it. I was already on the bpo.

**Comment by vstinner** [comment]
@pitrou: Does it look better? Now, it's hard to remove even more things from my PR!

**Comment by pitrou** [comment]
Thank you! Looks good to me now.

**Comment by vstinner** [comment]
"Thank you! Looks good to me now."

Thanks for the review @pitrou.

**Pull request merged**